src/fcio.c

```diff
@@ -449,6 +453,25 @@ Returns the internal FCIOStream object on success or NULL on error.
   return xio;
 }
 
+
+static inline int fcio_put_config(FCIOStream output, fcio_config* config)
```

Comment:
The fcio_put functions do not perform any input checks - we can either accept that and be very careful in their use (because they are static functions) or we add just two cheap checks `if (!output || !config) return` to be defensive.

Reply:
Added